### DISCUSSION TITLE
Re-export problem with format_description! when using time crate as a rank 2 dependency

### DISCUSSION THREAD

**Discussion by jphppd** [time-rs/time#525] [Question]
Dear maintainer,

Thank you for your work!

I have a re-export issue with [`format_description!`](https://docs.rs/time-macros/0.2.6/src/time_macros/lib.rs.html#80). In my project, `time` is only used by/through the [`simplelog`](https://github.com/drakulix/simplelog.rs) crate. Hence, I expect _not_ to have to add a dependency to `time` in my `Cargo.toml`.

Doing so, when I compile, I get this error:

```
error[E0433]: failed to resolve: could not find `time` in the list of imported crates
 --> src/main.rs:3:33
  |
3 |         .set_time_format_custom(simplelog::format_description!("[day]"))
  |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `time` in the list of imported crates

```

This error does **not** occur when I _do_ add the `time` dependency. I suppose that the problem originates in some way [here](https://docs.rs/time-macros/0.2.6/src/time_macros/lib.rs.html#86),
but macro are still a bit uncharted territory to me...

Apparently _(live help on Discord)_, a way to fix this would be to pass the crate name in the definition of the macro in `time-macros`, and use the magic variable `$crate` in `time` before re-exporting. Not sure of the impact though...

Do you have any comment, any suggestion, any fix?
It's not a blocking issue, but it's not a very nice behavior either.

Best

---

simplelog uses this macro [here](https://github.com/Drakulix/simplelog.rs/blob/v0.12.0/src/config.rs#L8), re-exports [there](https://github.com/Drakulix/simplelog.rs/blob/v0.12.0/src/lib.rs#L27)

Dependencies tree:
```
simplelog v0.12.0
├── ...
└── time v0.3.17
    ├── ...
    └── time-macros v0.2.6 (proc-macro)
        └── time-core v0.1.0
```

Minimum failing example:

`Cargo.toml`:
```toml
[dependencies]
simplelog = { version = "=0.12.0" }
```

`main.rs`:
```rust
fn main() {
    let loc_config_term = simplelog::ConfigBuilder::new()
        .set_time_format_custom(simplelog::format_description!("[day]"))
        .build();
}
```

**Comment by jhpratt** [comment]
Yes, passing the crate name to the macro is an option, but that is not a solution I am interested in. Until Rust gains a proper solution for declaring runtime dependencies, you should suggest that `time` as a direct dependency.
